### PR TITLE
add relay

### PR DESCRIPTION
**Pull request by rawdaGastan**
### Related Issues

- https://github.com/threefoldtech/grid3-go/issues/111

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -27,5 +27,5 @@ jobs:
       - name: Test
         env:
           MNEMONICS: ${{ secrets.MNEMONICS }}
-          NETWORK: ${{ secrets.NETWORK }}
+          NETWORK: dev
```
Comment (xmonader) [inline .github/workflows/integration.yml:33]: please make sure there's an issue exist to change that back in the future

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -24,5 +24,5 @@ jobs:
       - name: Test
         env:
           MNEMONICS: ${{ secrets.MNEMONICS }}
-          NETWORK: ${{ secrets.NETWORK }}
+          NETWORK: dev
```
Comment (xmonader) [inline .github/workflows/test.yml:27]: same as for integration.yml, we need an issue to track this

File: .github/workflows/integration.yml (hunk 1 above)
Comment (rawdaGastan) [reply .github/workflows/integration.yml:33]: - https://github.com/threefoldtech/grid3-go/issues/113

File: .github/workflows/test.yml (hunk 2 above)
Comment (rawdaGastan) [reply .github/workflows/test.yml:27]: - https://github.com/threefoldtech/grid3-go/issues/113

File: deployer/tf_plugin_client.go [hunk 3]
```diff
@@ -79,6 +90,7 @@ func NewTFPluginClient(
 	keyType string,
 	network string,
 	substrateURL string,
+	passedRelayURL string,
```
Comment (xmonader) [inline deployer/tf_plugin_client.go]: why not relayURL?

File: deployer/tf_plugin_client.go [hunk 4]
```diff
@@ -79,6 +90,7 @@ func NewTFPluginClient(
...
 	passedRmbProxyURL string,
```
Comment (xmonader) [inline deployer/tf_plugin_client.go]: I don't think we need this parameter anymore for proxying, but do we need it for grid rest API?

File: deployer/validation.go [hunk 5]
```diff
@@ -104,14 +56,14 @@ func validateMnemonics(mnemonics string) error {
 	return nil
 }
 
-func validateSubstrateURL(url string) error {
+func validateWssURL(url string) error {
 	if len(url) == 0 {
```
Comment (xmonader) [inline deployer/validation.go]: Let's add TrimSpace to it

File: deployer/validation.go [hunk 6]
```diff
@@ -104,14 +56,14 @@ func validateMnemonics(mnemonics string) error {
...
-		return errors.New("substrate url is required")
+		return errors.New("url is required")
 	}
 
 	alphaOnly := regexp.MustCompile(`^wss:\/\/[a-z0-9]+\.[a-z0-9]\/?([^\s<>\#%"\,\{\}\\|\\\^\[\]]+)?$`)
 	if !alphaOnly.MatchString(url) {
-		return errors.New("substrate url is not valid")
+		return errors.New("wss url is not valid")
```
Comment (xmonader) [inline deployer/validation.go]: let's include the url

File: deployer/tf_plugin_client.go [hunk 7]
```diff
@@ -196,3 +216,7 @@ func NewTFPluginClient(
 
 	return tfPluginClient, nil
 }
+
+func generateSessionID(twinID uint32) string {
```
Comment (xmonader) [inline deployer/tf_plugin_client.go]: where's the twinID param used?

File: integration_tests/network_test.go [hunk 8]
```diff
@@ -82,4 +84,13 @@ func TestNetworkDeployment(t *testing.T) {
 		assert.Error(t, err)
 	})
 
+	t.Run("test get public node", func(t *testing.T) {
+		publicNodeID, err := workloads.GetPublicNode(
```
Comment (xmonader) [inline integration_tests/network_test.go]: I wonder why this function `GetPublicNode` is part of the workloads package?

**Review by xmonader — CHANGES_REQUESTED** [review]
Great job add minor comments, but should be good to go! Thanks!

File: integration_tests/gatway_fqdn_test.go [hunk 9]
```diff
@@ -117,7 +110,6 @@ func TestGatewayFQDNDeployment(t *testing.T) {
 	err = tfPluginClient.NetworkDeployer.Cancel(ctx, &network)
 	assert.NoError(t, err)
 
-	_, err = tfPluginClient.State.LoadGatewayFqdnFromGrid(nodeID, gw.Name)
```
Comment (xmonader) [inline integration_tests/gatway_fqdn_test.go:120]: Let's make the function name `LoadGatewayFQDNFromGrid` instead of `LoadGatewayFqdnFromGrid`

**Review by xmonader — CHANGES_REQUESTED** [review]
Few comments were added, please address and merge

**Review by xmonader — APPROVED** [review]

**Pull request merged**